torch/csrc/jit/api/module.h

```diff
@@ -108,9 +108,7 @@ struct TORCH_API Module : public Object {
     return true;
   }
 
-  IValue forward(std::vector<IValue> inputs) {
-    return get_method("forward")(std::move(inputs));
-  }
+  IValue forward(std::vector<IValue> inputs);
```

Comment:
This is different from how Python handles hooks, it should probably be made so the interface is the same otherwise the hook behavior will be confusing.

Python:
```python
my_module(*inputs)  # hooks are run
my_module.forward(*inputs)  # hooks are not run
```

With this PR as-is:
```cpp
my_module.forward(inputs)  // hooks are run
```

`Module` should have an overload for `operator()` that calls the hooks and the `forward` method here should stay as it was before. So in the end it should look like:

```cpp
my_module(inputs)  // hooks are run
my_module.forward(inputs)  // hooks are not run
```

Reply:
Makes sense.